### ISSUE TITLE
4 error and 1 warning when run   luajit -l clnn -e 'clnn.test()' 

### ISSUE THREAD

**Issue by LaoAnchor** [hughperkins/clnn#34]
2016.3.16 i installed nn, cltorch and tested successfully.
Besides,i succeeding in running jcjohnson's code [](https://github.com/jcjohnson/neural-style) to get styled images.
but the test of clnn shows there still are 4 errors and a warning. I'm the newer and have no idea about the reason, can anyone explain it?

```
`65/70 LookupTable_backward .............................................. [PASS]
66/70 SpatialAveragePooling_forward_ceil ................................ [PASS]
67/70 SpatialMaxPooling_forward_batch ................................... [PASS]
68/70 Sqrt_forward ...................................................... [PASS]
69/70 Sum_backward ...................................................... [PASS]
70/70 LookupTable_forward ............................................... [PASS]
Completed 111 asserts in 70 tests with 0 failures and 4 errors and 1 warning
--------------------------------------------------------------------------------
ELU_backward
 Function call failed
/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: wrong number of arguments for function call
stack traceback:
	[C]: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: in function 'ELU_updateOutput'
	/home/gpu2/torch/install/share/lua/5.1/nn/ELU.lua:20: in function 'forward'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:200: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2619: in function </home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2617>
	[C]: in function 'xpcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:476: in function '_pcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:436: in function '_run'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:354: in function 'run'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2658: in function 'test'
	(command line):1: in main chunk
	[C]: at 0x00406670

--------------------------------------------------------------------------------
ELU_forward
 Function call failed
/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: wrong number of arguments for function call
stack traceback:
	[C]: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: in function 'ELU_updateOutput'
	/home/gpu2/torch/install/share/lua/5.1/nn/ELU.lua:20: in function 'forward'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:166: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2619: in function </home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2617>
	[C]: in function 'xpcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:476: in function '_pcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:436: in function '_run'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:354: in function 'run'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2658: in function 'test'
	(command line):1: in main chunk
	[C]: at 0x00406670

--------------------------------------------------------------------------------
ELU_transposed
 Function call failed
/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: wrong number of arguments for function call
stack traceback:
	[C]: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/nn/THNN.lua:109: in function 'ELU_updateOutput'
	/home/gpu2/torch/install/share/lua/5.1/nn/ELU.lua:20: in function 'forward'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:64: in function 'pointwise_transposed'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:216: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2619: in function </home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2617>
	[C]: in function 'xpcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:476: in function '_pcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:436: in function '_run'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:354: in function 'run'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2658: in function 'test'
	(command line):1: in main chunk
	[C]: at 0x00406670

--------------------------------------------------------------------------------
mse_variablebatchsize
 Function call failed
/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:83: Unexpected arguments passed to test function
stack traceback:
	[C]: in function 'assert'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:83: in function 'getMessage'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:126: in function 'assert'
	...u2/torch/install/share/lua/5.1/clnn/testMSECriterion.lua:122: in function 'v'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2619: in function </home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2617>
	[C]: in function 'xpcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:476: in function '_pcall'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:436: in function '_run'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:354: in function 'run'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2658: in function 'test'
	(command line):1: in main chunk
	[C]: at 0x00406670

--------------------------------------------------------------------------------
Should use TestSuite rather than plain lua table

--------------------------------------------------------------------------------
luajit: /home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:362: An error was found while running tests!
stack traceback:
	[C]: in function 'assert'
	/home/gpu2/torch/install/share/lua/5.1/torch/Tester.lua:362: in function 'run'
	/home/gpu2/torch/install/share/lua/5.1/clnn/test.lua:2658: in function 'test'
	(command line):1: in main chunk
	[C]: at 0x00406670
`
```

**Comment by hughperkins** [comment]
Hmmm, right. It seems that the torch `Tester.lua` library changed, in a non-backwards compatible way.

**Comment by hughperkins** [comment]
oh... not the only change either actually... :-(

**Comment by hughperkins** [comment]
ELU test errors addressed in 60954c9

**Comment by hughperkins** [comment]
MSECriterion error addressed in 3232a17
